libs/komenci-logger/src/komenci-logger.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, LoggerService } from '@nestjs/common'
+import { Logger } from "nestjs-pino"
+
+import {
+    AttestationsRequested,
+    KomenciEvent,
+    MetaTransactionSubmitted,
+    PepperRequested,
+    RuleVerified,
+    TxEvent,
+    WalletDeployed
+} from '@app/komenci-logger/events'
+
+export interface KomenciLogger extends LoggerService {
+    logSuccessfulSessionStart: (sessionStartFailEvent: KomenciEvent) => void
+    logFailedSessionStart: (sessionStartFailEvent: KomenciEvent) => void
+    logRuleChecked: (ruleVerifiedEvent: RuleVerified) => void
+    logWalletDeployed: (walletDeployEvent: WalletDeployed) => void
+    logBlindedPepperRequest: (pepperRequestEvent: PepperRequested) => void
+    logCompletedTransaction: (txEvent: TxEvent) => void
+    logSubmittedMetaTransaction: (submittedMTx: MetaTransactionSubmitted) => void
+    logSubsidizedAttestations: (attestationsRequested: AttestationsRequested) => void
+}
+
+@Injectable()
+export class KomenciLoggerService implements KomenciLogger {
+    constructor(private readonly logger: Logger) {}
+    log(message: any, context?: any, ...args): void {
+        this.logger.log(message, context, ...args)
+    }
+    verbose(message: any, context?: any, ...args): void {
+        this.logger.verbose(message, context, ...args)
+    }
+    debug(message: any, context?: any, ...args): void {
+        this.logger.debug(message, context, ...args)
+    }
+    warn(message: any, context?: any, ...args): void {
+        this.logger.warn(message, context, ...args)
+    }
+    error(message: any, context?: any, ...args): void {
+        this.logger.error(message, context, ...args)
+    }
+
+
+    logSuccessfulSessionStart(sessionStartEvent: KomenciEvent): void {
+        this.log("Session Started Successfully", sessionStartEvent)
+    }
+
+    logFailedSessionStart(sessionStartFailEvent: KomenciEvent): void {
+        this.log('Session Start Failed', sessionStartFailEvent)
+    }
+
+    logRuleChecked(ruleVerifiedEvent: RuleVerified): void {
+        this.log('Rule Verified', ruleVerifiedEvent)
+    }
+
+    logWalletDeployed(walletDeployEvent: WalletDeployed): void {
+        this.log('Wallet Deployed', walletDeployEvent)
+    }
+
+    logBlindedPepperRequest(pepperRequestEvent: PepperRequested): void {
+        this.log('Blinded Pepper Request', pepperRequestEvent)
+    }
+
+    logCompletedTransaction(txEvent: TxEvent): void {
+        this.log('Transaction Successful', txEvent)
+    }
+
+    logSubmittedMetaTransaction(submittedMTx: MetaTransactionSubmitted): void {
+        this.log('MetaTransaction Submitted', submittedMTx)
+    }
+    logSubsidizedAttestations(attestationsRequested: AttestationsRequested): void {
```

Comment:
This doesn't really make sense like this. Lets implement a `logEvent` that takes an `event: KEvent` as argument. 
I sketched out an implementation [here](https://www.typescriptlang.org/play?ssl=4&ssc=2&pln=1&pc=1#code/JYOwLgpgTgZghgYwgAgEJwM4QKIDcLgAKUA9gA4bIDeAUMvclhhsCSAJIAmA-AFyNgooAOZ0GEAB6QoIOABsAgggQkAruH4ZBImgF8aNOACMtURGGQI5mSgGk8BMAB4AKsXKVJkEJ0rosDkSkFMgAvGiYOPhBHgB8tPRkqkZywAjIxqbmyFAQcJxscgCeyLIAthCa2iDCANxiKiCmqghgJFAAFEkpaTl5BSDFyGTBGPxuowCU1Pr6NFY2yADKYHBQYEsQzKwggRZeBL7I9tEWCaVwFWHIAEQraxtbLGw3egag0vBIyADyYAAW0D27hCBx8fkiwNG1DEGBIFT+gKgADFgBA5Jx+CBVGUjNA3vNrMxfgCgadkGCjidHE5EWTHCCMLEYfRyihwjc6VA9q85jB1K0dsg5CRhHsOhBTvxqeAnHAQEVYtNzo04XIIAA6EXCCWnDVsgA0FL1Iw8kwJNG14rEIAgAHcSUjxed6ExnhxMbdMJwbgaxPQDjJ5EoVOowPwbt7ff7GPCIFzUejPQBGABMAGYxLpzebLaLraz7ctVutNts2M6Y26dlwI1G-QwKVJoLJFMo1BovRgffQszmaEA)